contrib/google.golang.org/grpc/grpc_test.go

```diff
@@ -262,48 +262,121 @@ func TestStreaming(t *testing.T) {
 	})
 }
 
-func TestChild(t *testing.T) {
-	assert := assert.New(t)
-	mt := mocktracer.Start()
-	defer mt.Stop()
-
-	rig, err := newRig(false)
-	if err != nil {
-		t.Fatalf("error setting up rig: %s", err)
+func TestSpanTree(t *testing.T) {
+	assertSpan := func(t *testing.T, span, parent mocktracer.Span, operationName, resourceName string) {
+		require.NotNil(t, span)
+		assert.Nil(t, span.Tag(ext.Error))
+		assert.Equal(t, operationName, span.OperationName())
+		assert.Equal(t, "grpc", span.Tag(ext.ServiceName))
+		assert.Equal(t, span.Tag(ext.ResourceName), resourceName)
+		assert.True(t, span.FinishTime().Sub(span.StartTime()) > 0)
+
+		if parent == nil {
+			return
+		}
+		assert.Equal(t, parent.SpanID(), span.ParentID(), "unexpected parent id")
 	}
-	defer rig.Close()
 
-	client := rig.client
-	resp, err := client.Ping(context.Background(), &FixtureRequest{Name: "child"})
-	assert.Nil(err)
-	assert.Equal(resp.Message, "child")
+	t.Run("unary", func(t *testing.T) {
+		assert := assert.New(t)
+		mt := mocktracer.Start()
+		defer mt.Stop()
 
-	spans := mt.FinishedSpans()
-	assert.Len(spans, 2)
+		rig, err := newRig(true)
+		if err != nil {
+			t.Fatalf("error setting up rig: %s", err)
+		}
+		defer rig.Close()
 
-	var serverSpan, clientSpan mocktracer.Span
+		{
```

Comment:
What is the reason for putting this in a block?

Reply:
The following `rootSpan` type is `tracer.Span` and then come the assertions with the tracer mockup spans which have different types. So I created a new separate scope to avoid having two types of spans in the function scope.

Happy to put it in a separate function if you prefer.